consonance-webservice/src/main/java/io/swagger/wes/api/impl/Ga4ghApiServiceImpl.java

```diff
@@ -0,0 +1,414 @@
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(Ga4ghApiServiceImpl.class);
+
+    private HierarchicalINIConfiguration settings = null;
+    private static ConsonanceWebserviceConfiguration config;
+    private static OrderResource orderResource;
+    private static Gson gson = new Gson();
+
+    public static void setConfig(ConsonanceWebserviceConfiguration config) {
+        Ga4ghApiServiceImpl.config = config;
+    }
+
+    public static void setOrderResource(OrderResource orderResource) {
+        Ga4ghApiServiceImpl.orderResource = orderResource;
+    }
+
+    // Cancels a running workflow by providing the id of the workflow
+    @Override
+    public Response cancelJob(String workflowId, ConsonanceUser user) throws NotFoundException {
+
+        return Response.ok().build();
+    }
+
+
+    //Provides information on the service, versions ...
+    @Override
+    public Response getServiceInfo(ConsonanceUser user) throws NotFoundException {
+
+        Ga4ghWesServiceInfo serviceInfo = new Ga4ghWesServiceInfo();
+        // Supported formats
+        String cwl = "cwl";
+        String wdl = "wdl";
+        // Version of each
+        String wdlVersion = "1.0"; // Latest release 2017-08-16T00:00:00.000-00:00
+        String cwlVersion = "1.0";
+
+        // Supported WDL version
+        Ga4ghWesWorkflowTypeVersion versionTypeWDL = new Ga4ghWesWorkflowTypeVersion();
+        versionTypeWDL.addWorkflowTypeVersionItem(wdlVersion);
+
+        // Add to Map the WDL specifics
+        serviceInfo.putWorkflowTypeVersionsItem(wdl, versionTypeWDL);
+
+        // Supported CWL version
+        Ga4ghWesWorkflowTypeVersion versionTypeCWL = new Ga4ghWesWorkflowTypeVersion();
+        versionTypeCWL.addWorkflowTypeVersionItem(cwlVersion);
+
+        // Add it to the map
+        serviceInfo.putWorkflowTypeVersionsItem(cwl, versionTypeCWL);
+
+        // WES version supported
+        String wesApiVersion = "v1.0";
+
+        // Add to map
+        serviceInfo.addSupportedWesVersionsItem(wesApiVersion);
+
+        // Supported file system protocol
+        List<String> supportedFileProtocols = Arrays.asList("http", "https", "sftp", "s3", "gs", "synapse");
+
+        // Add to map
+        serviceInfo.setSupportedFilesystemProtocols(supportedFileProtocols);
+
+        // Engine and version
+        String cwltool = "cwltool";
+        String cwltoolVersion = "1.0.20171107133715";
+        serviceInfo.putEngineVersionsItem(cwltool, cwltoolVersion);
+        String dockstore = "dockstore";
+        String dockstoreVersion = "1.3.0";
+        serviceInfo.putEngineVersionsItem(dockstore, dockstoreVersion);
+
+        // System state counts
+        Map <JobState, Long> systemState = new HashMap<JobState, Long>();
+
+        List<Job> allJobs = orderResource.listWorkflowRuns(user);
+        if (allJobs.isEmpty()){
+            serviceInfo.putSystemStateCountsItem("Unknown", (long) 0);
```

Comment:
Are there enums or something for this or line 168? 
0 and 1 seem arbitrary, although adding 1 on line 165 might be clever